### PR TITLE
Add the choice for inserting comma or semicolon automatically.

### PR DESCRIPTION
**Pull request by ammarnajjar**
This was suggested previously by @daGrevis in issue #4.
I find it useful for my own workflow, so here I share it.
It currently triggers inserting comma or semicolon when leaving insert mode on the same line only. It may a good start though.

### CODE REVIEW

File: plugin/cosco.vim [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        echo "AutoCommaOrSemiColon is OFF"
+    else
+        let g:auto_comma_or_semicolon = 1
+        echo "AutoCommaOrSemiColon is ON"
+    endif
+endfunction
+
+command! CommaOrSemiColon call cosco#commaOrSemiColon()
+function! AutoCommaOrSemiColon()
+    let b:currentLineLastChar = matchstr(getline(line('.')), '.$')
```
Comment (lfilho) [inline plugin/cosco.vim]: You can use just `getline('.')` intead of `getline(line('.'))` here.

Comment (lfilho) [inline plugin/cosco.vim]: Why do we need to grab the current line's last char before calling `cosco#commaOrSemiColon()` here?

Also, if that line is really necessary, i think we can move it to inside the `if` below for performances reasons (not executing those operations for getting the last chars unless it's really needed...

**Comment by ammarnajjar** [comment]
I was experimenting with "TextChanged" event. There still one is forced to go back to normal mode before moving to new line to get the correct behavior. I forgot to remove that line. sorry.

**Comment by lfilho** [comment]
Great job, @ammarnajjar. I left a couple of minor comments for you.

Also, I was thinking if an auto feature is something wise here. I mean: the plugin is not to be trusted 100% (we know it doesn't work for 100% of the cases), so having users trust the plugin blindly due the existence of such a feature sounds risky, something that we, as the providers/authors, should consider more carefully.

What do you think?

**Comment by ammarnajjar** [comment]
I agree with you, but as I said, I found it useful for me in my daily life, after moving from python to c++. (I get a brain tumor when the compiler complains about missing semicolon)
All in all, this is just experimental feature, needs improvements for sure, but as in all open-source software, someone should start somewhere.
I think  people might find the patch useful if they wanna take the risk for they know what they are doing.

**Comment by lfilho** [comment]
Yeah...
I think it's a good idea to have a note in the readme's part you wrote, warning the users about this risk. What do you think?

**Comment by ammarnajjar** [comment]
btw: personally: what made me look for this plugin in the first place is the need to auto insert semicolons. I don't want to  insert them manually, not even with a shortcut, so I think the full benefit is when it does the auto insertion correctly.
Good idea, I will add a warning to the related section.

**Comment by lfilho** [comment]
I agree with you. But I think it is impossible to auto-insert it correctly without a proper parser for each language, after all of them will have specifics and edge cases... So this plugin can only go so far... :)

Alright, I will wait for that last addition and tomorrow I'll check your branch out and give it a quick test and then merge it

**Comment by lfilho** [comment]
Cool. Seems to work fine.
But as expected, I won't turn it on for me. In a few seconds I already hit some edge cases in my javascript files 😅 .

Feel free to contribute more to the plugin to make it more robust 👍 
And thanks again for you contribution
